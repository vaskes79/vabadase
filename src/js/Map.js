import GoogleMapsLoader from 'google-maps';

class Map {
  constructor(rootSelector = '#gmap') {
    this.mapContainer = document.querySelector(rootSelector);
    this.center = [59.462475, 28.0408413];
    this.key = 'AIzaSyBFJkLTCugVE8sB4qEJ3-AXfFoGsFRvJ14';
    this.ver = '3.36';
  }

  init = () => {
    const {center, mapContainer, key, ver} = this;
    GoogleMapsLoader.KEY = key;
    GoogleMapsLoader.VERSION = ver;
    GoogleMapsLoader.load(google => {
      const options = {
        zoom: 17,
        center: new google.maps.LatLng(center[0], center[1]),
        mapTypeId: google.maps.MapTypeId.ROADMAP,
      };
      new google.maps.Map(mapContainer, options);
    });
  };
}

export default Map;
