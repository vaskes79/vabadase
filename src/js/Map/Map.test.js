import Map from './Map';

let map = null;
let mock = null;
let clickHandlerMockL = null;

beforeEach(() => {
  map = new Map();
  // clickHandlerMockL = jest.spyOn(map, 'handlerBtnL');
  map.init();
  mock = jest.fn();
});

afterEach(() => {
  map = null;
  mock = null;
  clickHandlerMockL = null;
});

describe(`Create instance and render`, () => {
  it(`expect that map instance of Map`, () => {
    expect(map instanceof Map).toBeTruthy();
  });

  it(`render components into DOM`, () => {
    expect(map).toMatchSnapshot();
  });
});
