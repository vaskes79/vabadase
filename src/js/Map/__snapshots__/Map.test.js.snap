// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create instance and render render components into DOM 1`] = `
Map {
  "center": Array [
    59.462475,
    28.0408413,
  ],
  "init": [Function],
  "key": undefined,
  "lang": undefined,
  "mapContainer": <div
    class="contacts__map"
    id="gmap"
  />,
  "ver": undefined,
}
`;
